package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html

```diff
@@ -64,6 +64,19 @@
 		return v and translate('enabled') or translate('disabled')
 	end
 
+	local function formatBits(v, factor)
+		--to format a value given in kilobits use a factor of 1000
+		factor = factor or 1
+		units = {[0]='', 'k', 'M', 'G'}
+
+		bits = math.max(0, v*factor)
```

Comment:
The math.max is unneeded here as you said that the data comes from respondd and can't be negative... It is unexpected behavior to get 0 when you input a negative value.

Furthermore I'm now even more convinced that you should omit the "factor" parameter as you could just call the function as
`formatBits(bits * factor)` instead of `formatBits(bits, factor)`. Having factor as a parameter just adds to complexity and confusion...

Reply:
I feel like I'd reduce code quality if I removed such checks, just because I know what one data provider acts like.
Lets call it a weekend. Maybe another pair of eyes should decide on those enhancements you proposed?

Reply:
I don't think you would reduce code quality, but the opposite. There may be use-cases where negative values might be useful. Handling negative values as 0 is not a good solution.

Reply:
Why do you think having "factor" as a second parameter is wishful if it just does a multiplication while it is possible to do the multiplication beforehand with less code and better readability? I don't understand why you think anyone experienced in programming would have a different opinion on that...

Reply:
```suggestion
		bits = v*factor
```

Together with the other suggestion, this should be fine now.

Reply:
http://lilienstra-e.n.ffh.zone/

Reply:
negative values work, there's only one paramter, zero values work now.